pkg/k8s/apis/cilium.io/v2/register.go

```diff
@@ -112,16 +113,41 @@ func CreateCustomResourceDefinitions(clientset apiextensionsclient.Interface) er
 				ShortNames: []string{"cnp", "ciliumnp"},
 				Kind:       CustomResourceDefinitionKind,
 			},
-			Scope: apiextensionsv1beta1.NamespaceScoped,
+			Scope:      apiextensionsv1beta1.NamespaceScoped,
+			Validation: &crv,
 		},
 	}
 
-	_, err := clientset.ApiextensionsV1beta1().CustomResourceDefinitions().Create(res)
+	log.Info("Creating v2.CiliumNetworkPolicy CustomResourceDefinition...")
+	clusterCRD, err := clientset.ApiextensionsV1beta1().CustomResourceDefinitions().Create(res)
 	if err != nil && !errors.IsAlreadyExists(err) {
 		return err
 	}
+	// If the CRD already exists on the cluster but without any validation then
+	// we need to create it.
+	if errors.IsAlreadyExists(err) && clusterCRD.Spec.Validation == nil {
+		// wait for CRD being established
+		err = wait.Poll(500*time.Millisecond, 60*time.Second, func() (bool, error) {
+			clusterCRD, err = clientset.ApiextensionsV1beta1().CustomResourceDefinitions().Get(cnpCRDName, metav1.GetOptions{})
+			if err != nil {
+				return false, err
```

Comment:
Won't this break the Poll call? Is it possible for the CRD to not yet be available via API and apiserver returns 404, the Poll will finish due to error. Is it expected behaviour? If yes, please consider changing the comment on line 129, because it suggests that we are polling apiserver until CRD is available (at least IMO).

Reply:
> Won't this break the Poll call? Is it possible for the CRD to not yet be available via API and apiserver returns 404, the Poll will finish due to error. Is it expected behaviour? 

The CRD is created on line 122. Here, we are updating an existing cilium CRD that doesn't contain the validation.